Mr robot

NOEMOJI041 opened the pull request:
MR ROBOT WIKI
ADDED :-
1) OVERVIEW
2) CODE EXPLANATION
3) MISCELLANEOUS

Kartik9250 commented on markdown/projects/mr-robot/major-code/arduino.rst:
```diff
@@ -0,0 +1,113 @@
+
+    }
+
+This code snippet is used to set up the various peripherals and settings needed for the robot to operate, including the ROS node, PWM signals for the motors, encoder signals for the wheels, and interrupt service routines for updating the encoder counts. 
```
Peripherals is not the right word to be used to address the components/parts here

Kartik9250 commented on markdown/projects/mr-robot/major-code/arduino.rst:
```diff
@@ -0,0 +1,113 @@
+        if(pwm.data < 0)  { ledcWrite(ledChannel1, 0); ledcWrite(ledChannel2, abs(pwm.data));}
+    }
+
+This code defines two callback functions, LpwmCb and RpwmCb, which take a std_msgs::Int32 object as input. The functions control the speed of four motors by writing PWM signals using the ledcWrite function.
```
Four motors were not used. Write two instead.

Kartik9250 replied:
Add Inline code for highlighting the different code components: LpwmCb, std_msgs

Kartik9250 commented on markdown/projects/mr-robot/major-code/arduino.rst:
Define how arduino is communicating the Ros master node

Kartik9250 commented on markdown/projects/mr-robot/major-code/arduino.rst:
```diff
@@ -0,0 +1,113 @@
+      right_enc_error.publish(&encoder_msg_right_error);
+    }
+
+The function publish_encoder_data() is used to publish the encoder values and any associated errors to their respective ROS topics.
```
inline coments

Kartik9250 reviewed (changes requested):
Use the following tag in order to make it resourceful & easier to read:

- Inline code tags
- Notes, Warnings, Seealso
- Link and cross reference our documentation pages.
- Use the Additional images that were added.
- Proof read the documentation and verify the provided information.
- Add references for the various websites for detailed information.

jasmeet0915 commented:
Some changes are required @NOEMOJI041 :
cc: @Kartik9250 @namikxgithub 
 - Mention MR Robot fullform in Project intro(overview) page and it should be written as MR Robot and not Mr. Robot since it is an Acronym.
- Add image captions 
 - Add official Links for packages and hardware components in Working section
 - Add tf tree for the robot in working section
 - Include ros computation graph of nodes in launching section to show topics and nodes that are launched. This can be done using [rqt_graph](http://wiki.ros.org/rqt_graph) node.
 - Doesn't need such big description for Master-Slave. We already cover this in ROS Intro. This section should be named different and it should contain a block diagram showing where the different nodes of the setup run physically. This block diagram is a different from ROS computation graph and not necessarily needs to include topics. But should show where different nodes run (pi, laptop/pc etc.)

NOEMOJI041 commented:
Fixed:-
1)Mr. Robot ------> MR Robot
2)Information Check
3)Deleted Extra from Master-Slave and changed its name
Added:-
1)inline code tags
2)Image Captions
3)Official Links in working.rst
4)Note is added in esp-diff-tf.rst and Warning is added to bash.rst
Note:-
tf tree and rqt_graph will be added according to the latest test.

NOEMOJI041 commented:
Added TF TREE and Node Graph

Kartik9250 commented on markdown/projects/mr-robot/mr-index.rst:
```diff
@@ -0,0 +1,14 @@
+========
+
+
+Mr robot is autonomous navigation robot made by A.T.O.M Robotics capable of doing multiple day to day operations such as mapping, navigation for transportation, sanitaion etc. Various other operations can also be performed thanks to its modularity.
```
Mr robot is an* autonomous

NOEMOJI041 replied:
done

Kartik9250 reviewed (approved):
Minor changes required, would be updated in next ver
